pkg/spoke/controllers/cachecontroller/executor_cache_controller.go

```diff
@@ -0,0 +1,318 @@
+package cachecontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	rbacapiv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	rbacv1 "k8s.io/client-go/informers/rbac/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	worklister "open-cluster-management.io/api/client/work/listers/work/v1"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+	"open-cluster-management.io/work/pkg/helper"
+	"open-cluster-management.io/work/pkg/spoke/auth"
+)
+
+var (
+	ResyncInterval = 10 * time.Minute
+)
+
+// CacheController is to reconcile the executor auth result for manfiestwork workloads
+// on spoke cluster.
+type CacheController struct {
+	manifestWorkLister worklister.ManifestWorkNamespaceLister
+	executorCaches     *auth.ExecutorCaches
+	sarCheckerFn       auth.SubjectAccessReviewCheckFn
+	restMapper         meta.RESTMapper
+}
+
+// NewExecutorCacheController returns an ExecutorCacheController
+func NewExecutorCacheController(
+	ctx context.Context,
+	recorder events.Recorder,
+	manifestWorkLister worklister.ManifestWorkNamespaceLister,
+	crbInformer rbacv1.ClusterRoleBindingInformer,
+	rbInformer rbacv1.RoleBindingInformer,
+	crInformer rbacv1.ClusterRoleInformer,
+	rInformer rbacv1.RoleInformer,
+	restMapper meta.RESTMapper,
+	executorCaches *auth.ExecutorCaches,
+	sarCheckerFn auth.SubjectAccessReviewCheckFn,
+) factory.Controller {
+
+	err := crbInformer.Informer().AddIndexers(
+		cache.Indexers{
+			byClusterRole: crbIndexByClusterRole,
+		},
+	)
+	if err != nil {
+		utilruntime.HandleError(err)
+	}
+
+	err = rbInformer.Informer().AddIndexers(
+		cache.Indexers{
+			byClusterRole: rbIndexByClusterRole,
+			byRole:        rbIndexByRole,
+		},
+	)
+	if err != nil {
+		utilruntime.HandleError(err)
+	}
+
+	controller := &CacheController{
+		manifestWorkLister: manifestWorkLister,
+		restMapper:         restMapper,
+		executorCaches:     executorCaches,
+		sarCheckerFn:       sarCheckerFn,
+	}
+
+	return factory.New().
+		WithInformersQueueKeysFunc(
+			roleEnqueueFu(rbInformer.Informer().GetIndexer(), executorCaches),
+			rInformer.Informer()).
```

Comment:
we might need to use specific AddEventHandler with bare informer here. I think it is easy to handle add/update event, but how about delete event?